tests/phpunit/tests/Checker/Checks/Plugin_Readme_Check_Tests.php

```diff
@@ -242,6 +242,103 @@ public function test_run_with_errors_parser_warnings() {
 		$this->assertEquals( 'readme_parser_warnings', $warnings['readme.txt'][0][0][0]['code'] );
 	}
 
+	public function test_run_with_errors_multiple_parser_warnings() {
+		$readme_check  = new Plugin_Readme_Check();
+		$check_context = new Check_Context( UNIT_TESTS_PLUGIN_DIR . 'test-plugin-plugin-readme-multiple-parser-warnings/load.php' );
+		$check_result  = new Check_Result( $check_context );
+
+		$readme_check->run( $check_result );
+
+		$errors   = $check_result->get_errors();
+		$warnings = $check_result->get_warnings();
+
+		$this->assertNotEmpty( $warnings );
+		$this->assertArrayHasKey( 'readme.txt', $warnings );
+		$this->assertEquals( 6, $check_result->get_warning_count() );
+		$this->assertEmpty( $errors );
+		$this->assertEquals( 0, $check_result->get_error_count() );
+
+		// Check for parser warnings.
+		$this->assertArrayHasKey( 0, $warnings['readme.txt'] );
+		$this->assertArrayHasKey( 0, $warnings['readme.txt'][0] );
+
+		for ( $i = 0; $i < 6; ++$i ) {
+			$this->assertArrayHasKey( 'code', $warnings['readme.txt'][0][0][ $i ] );
+			$this->assertEquals( 'readme_parser_warnings', $warnings['readme.txt'][0][0][ $i ]['code'] );
+		}
+	}
+
+	public function test_run_with_errors_parser_warnings_and_http_disabled() {
```

Comment:
The `get_wordpress_stable_version` code needs adjustments if we want to test the RCs and Beta versions. We can handle that in follow-up enhancement work, as it's not a big deal.

Instead, let's remove this test, as it doesn't do anything different. In its place, we can add the following test.

```
public function test_run_with_errors_parser_warnings_with_custom_set_transient_version() {

	$version = '5.0';
	set_transient( 'wp_plugin_check_latest_wp_version', $version );

	$readme_check  = new Plugin_Readme_Check();
	$check_context = new Check_Context( UNIT_TESTS_PLUGIN_DIR . 'test-plugin-plugin-readme-parser-warnings/load.php' );
	$check_result  = new Check_Result( $check_context );

	$readme_check->run( $check_result );

	$errors   = $check_result->get_errors();
	$warnings = $check_result->get_warnings();

	$this->assertEmpty( $errors );
	$this->assertNotEmpty( $warnings );
	$this->assertEquals( 0, $check_result->get_error_count() );
	$this->assertEquals( 1, $check_result->get_warning_count() );

	$this->assertArrayHasKey( 0, $warnings['readme.txt'] );
	$this->assertArrayHasKey( 0, $warnings['readme.txt'][0] );
	$this->assertArrayHasKey( 'code', $warnings['readme.txt'][0][0][0] );
	$this->assertEquals( 'readme_parser_warnings', $warnings['readme.txt'][0][0][0]['code'] );
	$this->assertStringContainsString( 'The "Tested up to" field was ignored. This field should only contain a valid WordPress version such as "' . $version . '"', $warnings['readme.txt'][0][0][0]['message'] );

	delete_transient( 'wp_plugin_check_latest_wp_version' );
}
```

This test checks the custom version set up by the transient.